packages/e2e-tests/test-applications/nextjs-13-app-dir/tests/transactions.test.ts

```diff
@@ -0,0 +1,95 @@
+import { test, expect } from '@playwright/test';
+import { waitForTransaction } from '../../../test-utils/event-proxy-server';
+import axios, { AxiosError } from 'axios';
+
+const authToken = process.env.E2E_TEST_AUTH_TOKEN;
+const sentryTestOrgSlug = process.env.E2E_TEST_SENTRY_ORG_SLUG;
+const sentryTestProject = process.env.E2E_TEST_SENTRY_TEST_PROJECT;
+const EVENT_POLLING_TIMEOUT = 30_000;
+
+test('Sends a pageload transaction', async ({ page }) => {
+  const pageloadTransactionEventPromise = waitForTransaction('nextjs-13-app-dir', transactionEvent => {
+    return transactionEvent?.contexts?.trace?.op === 'pageload' && transactionEvent?.transaction === '/';
+  });
+
+  await page.goto('/');
+
+  const transactionEvent = await pageloadTransactionEventPromise;
+  const transactionEventId = transactionEvent.event_id;
+
+  console.log(`Polling for pageload transaction eventId: ${transactionEventId}`);
+
+  await expect
+    .poll(
+      async () => {
+        try {
+          const response = await axios.get(
+            `https://sentry.io/api/0/projects/${sentryTestOrgSlug}/${sentryTestProject}/events/${transactionEventId}/`,
+            { headers: { Authorization: `Bearer ${authToken}` } },
+          );
+
+          return response.status;
+        } catch (e) {
+          if (e instanceof AxiosError && e.response) {
+            if (e.response.status !== 404) {
+              throw e;
+            } else {
+              return e.response.status;
+            }
+          } else {
+            throw e;
+          }
+        }
+      },
+      {
+        timeout: EVENT_POLLING_TIMEOUT,
+      },
+    )
+    .toBe(200);
+});
+
+test('Sends a transaction for a server component', async ({ page }) => {
+  const serverComponentTransactionPromise = waitForTransaction('nextjs-13-app-dir', transactionEvent => {
+    return (
+      transactionEvent?.contexts?.trace?.op === 'function.nextjs' &&
+      transactionEvent?.transaction === 'Page Server Component (/user/[id])'
+    );
+  });
+
+  await page.goto('/user/4');
+
+  console.log('Waiting for server component transaction.');
+
+  const transactionEvent = await serverComponentTransactionPromise;
+  const transactionEventId = transactionEvent.event_id;
+
+  console.log(`Polling for server component transaction eventId: ${transactionEventId}`);
```

Comment:
m: Could we have `waitForTransaction` emit these logs? I'd like to keep the test surface as clean as possible.

Reply:
That in particular won't work since we use `waitForTransaction` to get the `transactionEventId`. If it is about the logging statement I can just remove it.

Reply:
But we can just log it out once the `serverComponentTransactionPromise` resolves internally?

Ideally the tests themselves have no logs. And if it does, it's optional debug logging that can be turned on.